### PR TITLE
Scratch UI

### PR DESCRIPTION
**Pull request by namdaoduy**
Demo Video in Slack

### CODE REVIEW

File: frontend/src/Components/ScratchIframe.tsx [hunk 1]
```diff
@@ -1,40 +1,119 @@
...
+  }
 
-    React.useEffect( () => {
-        const interval = setInterval(() => {
-            updateNewScratchUrl();
-        }, 1000);
-        return () => clearInterval(interval);
-    })
+  function onReload() {
+    setIframeKey(uuidv4());
```
Comment (vudung45) [inline frontend/src/Components/ScratchIframe.tsx:35]: is this required?

File: frontend/src/Components/ScratchIframe.tsx [hunk 2]
```diff
@@ -1,40 +1,119 @@
...
+  // React.useEffect(() => {
+  //   updateNewScratchUrl();
+  // }, [])
+
+  return (
+    <div className="u-border u-heightFull u-widthFull u-flex u-positionRelative u-backgroundWhite">
+      {!!scratchUrl ? (
+        <iframe
+          width="100%" height="100%"
+          key={iframeKey}
```
Comment (vudung45) [inline frontend/src/Components/ScratchIframe.tsx:48]: ^
Just out of curiosity, couldn't we just add a constant key here or maybe just get a random uuid once and use it for the rest of the sesson?

File: frontend/src/Components/ScratchIframe.tsx [hunk 3]
```diff
@@ -1,40 +1,119 @@
...
 
 
-export default function YoutubeIframe() {
-    let [scratchUrl, setScratchUrl] = React.useState();
+export default function ScratchIframe({
+  onCloseWindow,
+} : {
+  onCloseWindow: Function,
+}) {
+  let [scratchUrl, setScratchUrl] = React.useState();
```
Comment (vudung45) [inline frontend/src/Components/ScratchIframe.tsx:18]: NIT: 
```suggestion
  let [scratchUrl, setScratchUrl] = React.useState("");
```
for typescript to enforce typestring

File: frontend/src/Components/ScratchIframe.tsx (hunk 1 above)
Comment (namdaoduy) [reply frontend/src/Components/ScratchIframe.tsx:35]: @vudung45 
Changing the key will force the iframe to refresh. Random number or new uuid will both work in this case.

File: frontend/src/Components/ScratchIframe.tsx (hunk 2 above)
Comment (namdaoduy) [reply frontend/src/Components/ScratchIframe.tsx:48]: @vudung45 Answered above.

File: frontend/src/Components/ScratchIframe.tsx (hunk 1 above)
Comment (namdaoduy) [reply frontend/src/Components/ScratchIframe.tsx:35]: TODO: `await updateNewScratchUrl()` before `setIframeKey(uuidv4());`

Comment (namdaoduy) [reply frontend/src/Components/ScratchIframe.tsx:35]: Just to make sure the iframe won't refresh twice if scratchURL have new update.

**Pull request merged**